test/e2e/es/stack_monitoring_test.go

```diff
@@ -0,0 +1,155 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package es
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"testing"
+
+	esClient "github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/client"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/stackmon"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch"
+)
+
+// TestESStackMonitoring tests that when an Elasticsearch cluster is configured with monitoring, its log and metrics are
+// correctly delivered to the referenced monitoring Elasticsearch clusters.
+// It tests that the monitored ES pods have 3 containers ready and that there are documents indexed in the beat indexes
+// of the monitoring Elasticsearch clusters.
+func TestESStackMonitoring(t *testing.T) {
+	// only execute this test on supported version
+	err := stackmon.IsSupportedVersion(test.Ctx().ElasticStackVersion)
+	if err != nil {
+		t.SkipNow()
+	}
+
+	// create 1 monitored and 2 monitoring clusters to collect separately metrics and logs
+	metrics := elasticsearch.NewBuilder("test-es-mon-metrics").
+		WithESMasterDataNodes(2, elasticsearch.DefaultResources)
+	logs := elasticsearch.NewBuilder("test-es-mon-logs").
+		WithESMasterDataNodes(2, elasticsearch.DefaultResources)
+	monitored := elasticsearch.NewBuilder("test-es-mon-a").
+		WithESMasterDataNodes(1, elasticsearch.DefaultResources).
+		WithMonitoring(metrics.Ref(), logs.Ref())
```

Comment:
Not only, it's also because we're adding the namespace.

<clusterName>-es-monitoring-<namespace>-<clustername>-ca


36+15+63+1+36+3=154 < 253

test-es-mon-a-8xhw-es-monitoring-e2e-mercury-test-es-mon-logs-jbh2-ca


[
	{FieldValueInvalid Invalid value: "test-es-mon-a-8xhw-es-monitoring-e2e-mercury-test-es-mon-logs-jbh2-ca": must be no more than 63 characters spec.volumes[14].name} 
	{FieldValueInvalid Invalid value: "test-es-mon-a-8xhw-es-monitoring-e2e-mercury-test-es-mon-metrics-884m-ca": must be no more than 63 characters spec.volumes[15].name}

Reply:
My bad, I've not retested and I thought that the max volume name length was like the max secret name length (253) but it's 63.

The pattern used is: `<clusterName>-es-monitoring-<namespace>-<clustername>-ca` (ex: test-es-mon-a-8xhw-es-monitoring-e2e-mercury-test-es-mon-logs-jbh2-ca).
Max cluster name length is 36. Max namespace name length is 63.
36+15+63+1+36+3 = 154 > 63

So, we can't reuse the assoc ca secret name for the CA volume name. A volume name is 'local' to a pod, so we don't need the current es cluster name but we want to differentiate volumes coming from the different ES referenced for monitoring.
We could add the beginning of the sha of the "namespace+name" and still add the es ref name for debugging: `es-monitoring-<ns/clustername hash>-<clustername>-ca`?

Same issue with the config volume name:
`fmt.Sprintf("%s-%s-%s-config", resource.GetName(), assoc.AssociationType(), beatName)`
=> `<clusterName>-es-monitoring-metricbeat-config` 36+32 = 68 > 63

Reply:
Sounds good to me. Even just the hash I guess you would still see where it comes from in the referenced secret's name.

Reply:
c26faa5